adapters/limelightDigital/limelightDigital.go

```diff
@@ -0,0 +1,172 @@
+		if imp.BidFloor > 0 && imp.BidFloorCur != "" && strings.ToUpper(imp.BidFloorCur) != "USD" {
+
+			// Convert to US dollars
+			convertedValue, err := requestInfo.ConvertCurrency(imp.BidFloor, imp.BidFloorCur, "USD")
+			if err != nil {
+				return nil, []error{err}
+			}
+
+			// Update after conversion. All imp elements inside request.Imp are shallow copies
+			// therefore, their non-pointer values are not shared memory and are safe to modify.
+			imp.BidFloorCur = "USD"
+			imp.BidFloor = convertedValue
+		}
+
+		requestCopy.ID = request.ID + "-" + imp.ID
+		requestCopy.Imp = []openrtb2.Imp{imp}
+
+		requestJSON, err := json.Marshal(requestCopy)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requestData := &adapters.RequestData{
+			Method: "POST",
+			Uri:    url,
+			Body:   requestJSON,
+		}
+		requests = append(requests, requestData)
+	}
+	return requests, errors
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
+
+	if len(responseData.Body) == 0 {
+		return nil, nil
+	}
+
+	var response openrtb2.BidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+	bidResponse.Currency = response.Cur
+	for _, seatBid := range response.SeatBid {
+		for i, bid := range seatBid.Bid {
+			b := &adapters.TypedBid{
+				Bid:     &seatBid.Bid[i],
+				BidType: getMediaTypeForBid(bid.ImpID, request.Imp),
+			}
+			bidResponse.Bids = append(bidResponse.Bids, b)
+		}
+	}
+	return bidResponse, nil
+}
+
+func getImpressionExt(imp *openrtb2.Imp) (*openrtb_ext.ImpExtLimelightDigital, error) {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: "ext.bidder is not provided",
+		}
+	}
+	var limelightDigitalExt openrtb_ext.ImpExtLimelightDigital
+	if err := json.Unmarshal(bidderExt.Bidder, &limelightDigitalExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: "ext.bidder is not provided",
+		}
+	}
+	imp.Ext = nil
+	return &limelightDigitalExt, nil
+}
+
+func (a *adapter) buildEndpointURL(params *openrtb_ext.ImpExtLimelightDigital) (string, error) {
```

Comment:
This function need not be defined on adapter. This can be done as follow:-

`func buildEndpointURL(params *openrtb_ext.ImpExtLimelightDigital) (string, error)`

Reply:
I used `endpointTemplate` adapter param on 153 line. I rewrite it and receive template as argument in https://github.com/prebid/prebid-server/pull/2539/commits/b3d20782a84aaa4c9c184bbfb0dd9c896d56e7b9 commit. Thanks

Reply:
I'm not sure about this change. For other adapters  `buildEndpointURL` gets an adapter as a receiver. Please refer to usages of this function in other adapters.

Reply:
@VeronikaSolovei9 Thank you. I revert these changes in https://github.com/prebid/prebid-server/pull/2539/commits/4af1869e6e0d25e15621ec663aa6241e25b43cda commit. Thanks